AICopilot/socket_server.py

```
@@ -284,47 +301,149 @@ def _server_loop(self):
+                # matter how much more arrives, so that case (and a stalled
+                # peer) is bounded by the socket timeout above and the size
+                # cap below rather than looping forever.
+                buf = b""
+                decoder = json.JSONDecoder()
+                message = None
+                oversized = False
+                timed_out = False
+                while True:
+                    try:
+                        chunk = client_socket.recv(self._RECV_CHUNK_BYTES)
+                    except socket.timeout:
```

Ruff (0.16.1)

[warning] Replace aliased errors with `TimeoutError`

Replace `socket.timeout` with builtin `TimeoutError`

(UP041)